src/html/dom-module-name-or-is.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as dom5 from 'dom5';
+import {ParsedHtmlDocument} from 'polymer-analyzer/lib/html/html-document';
+import stripIndent = require('strip-indent');
+import {Warning, Severity} from 'polymer-analyzer/lib/warning/warning';
+import {stripWhitespace} from '../util';
+import {HtmlRule} from './rule';
+
+const p = dom5.predicates;
+
+export class DomModuleNameOrIs extends HtmlRule {
+  code = 'dom-module-invalid-attrs';
+  description = stripIndent(`
+      Warns for:
+
+          <dom-module name="foo-elem">
+          </dom-module>
+
+      or
+
+          <dom-module is="foo-elem">
+          </dom-module>
+
+      Correct syntax:
+
+          <dom-module id="foo-elem">
+          </dom-module>
+  `);
+  constructor() {
+    super();
+  }
+
+  async checkDocument(document: ParsedHtmlDocument) {
+    const warnings: Warning[] = [];
+    const badModule = p.AND(
+        p.hasTagName('dom-module'), p.OR(p.hasAttr('is'), p.hasAttr('name')));
+    for (const domModule of dom5.nodeWalkAll(document.ast, badModule)) {
```

Comment:
Sure.

My argument in favor of this form is that it makes it clear that the result isn't used anywhere else. It's like `const` only more so.